java/xtreemfs-servers/src/main/java/org/xtreemfs/mrc/osdselection/FileNamePrefixPolicy.java

```diff
@@ -0,0 +1,97 @@
+package org.xtreemfs.mrc.osdselection;
+
+import org.xtreemfs.common.uuids.ServiceUUID;
+import org.xtreemfs.mrc.metadata.XLocList;
+import org.xtreemfs.pbrpc.generatedinterfaces.DIR;
+import org.xtreemfs.pbrpc.generatedinterfaces.GlobalTypes;
+
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * OSDSelectionPolicy that allows to choose an OSD based on the path of the
+ * currently handled file.
+ */
+public class FileNamePrefixPolicy implements OSDSelectionPolicy {
+
+    public static final short POLICY_ID = (short) GlobalTypes
+            .OSDSelectionPolicyType
+            .OSD_SELECTION_POLICY_FILENAME_PREFIX
+            .getNumber();
+
+    /*
+     identifier for setting prefixes and OSD ids
+    */
+    private static final String attributeKeyString = "filenamePrefix";
+
+    /*
+      prefixes with an explicit OSD
+     */
+    private HashMap<String, String> prefixToOSDMap =
```

Comment:
The aim of this policy is to control OSD locations for folders explicitly. 
Hence automatic OSD selection for replication is not an expected use case of this policy.
Explicit OSD selection for replicas is not affected; also, you may specify a new OSD for the folder (prefix) before starting replication.
Also note that getOSDs(...) will never return an empty set of OSDs (except when the parameter allOSDs contains no OSDs, of course.)